search-sentiments-map/public/scripts/map-script.js

```diff
@@ -58,27 +58,41 @@ function initMap() {
 
 /** Loads the country boundary polygons from a GeoJSON source. */
 function loadMapOutline() {
-  // Load country data after finished loading in geojson.
-  map.data.loadGeoJson('countries.geojson', null, function () {
-    setNewTrend();
-  });
+  map.data.loadGeoJson('countries.geojson', null);
+}
+
+/** 
+ * Gets the selected mode (sentiment or popularity) from the webpage and loads
+ * data correspondingly.
+ */
+function loadCountryDataByMode() {
+  let sentimentMode = !document.getElementById('sentiment-popularity-check').checked;
+  console.log('sentiment mode', sentimentMode);
+
+  const topicHeader = document.getElementById('topic-header');
+  topicHeader.innerText = sentimentMode ?
+      'Worldwide sentiments of search results for "' + getCurrentTrend() + '"' :
+      'Worldwide interests for "' + getCurrentTrend() + '"' ;
+
+  loadCountryData(sentimentMode);
 }
 
-/** Loads the country sentiment score from Datastore. */
-function loadCountryData() {
-  map.data.forEach(function (row) {
-    const countryCode = row.getId();
-    let topicData = getCurrentCustomSearchData();
-    let countryData = topicData.countries
-      .filter(countries => countries.country === countryCode);
+/** 
+ * Loads the sentiments or search interests for all countries from Datastore. 
+ * @param {boolean=} sentimentMode Whether the result to obtain is the sentiments 
+ * (search interest otherwise).
```

Comment:
```suggestion
 * @param {boolean=} sentimentMode Whether the result to obtain is the sentiment data. Search Interest data will be loaded otherwise.
```

Reply:
ok